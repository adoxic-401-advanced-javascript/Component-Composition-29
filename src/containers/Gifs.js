import React, { Component } from 'react';
import Deck from '../components/Deck';

export default class Gifs extends Component {

  state = {
    number: 10,
    image: []
  }

  render() {
    return (
      <>
        <form onSubmit={this.handleSubmit}>
          <input
            type='number'
            name='number'
            value={this.state.number}
            onChange={this.handleNumberChange}
          />
          <button>Submit</button>
        </form>
        <Deck title="Gifs">
          {this.state.image}
        </Deck>
      </>
    );
  }

  handleSubmit = event => {
    event.preventDefault();
    this.apiCall();
  }

  handleNumberChange = ({ target }) =>  {
    this.setState({ number: +target.value });
  }

  componentDidMount() {
    this.apiCall();
  }

  apiCall = () => {
    return fetch(`https://hey-arnold-api.herokuapp.com/api/v1/gifs/random?count=${this.state.number}`)
      .then(res => res.json())
      .then((image) => {
        this.setState({ image }); 
      });
  }
}
