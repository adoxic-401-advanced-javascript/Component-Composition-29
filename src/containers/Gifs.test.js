import React from 'react';
import { shallow, mount } from 'enzyme';
import Gifs from './Gifs';

Gifs.apiCall = jest.fn();

describe('Gifs container', () => {
  it.skip('renders Gif component', () => {
    const wrapper = shallow(<Gifs />);
    expect(wrapper).toMatchSnapshot();
  });
});