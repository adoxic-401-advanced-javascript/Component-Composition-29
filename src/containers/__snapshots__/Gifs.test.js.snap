// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gifs container renders Gif component 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <input
      name="number"
      onChange={[Function]}
      type="number"
      value={10}
    />
    <button>
      Submit
    </button>
  </form>
  <Deck
    title="Gifs"
  />
</Fragment>
`;
